test/e2e/medusa_test.go

```diff
@@ -195,6 +196,18 @@ func verifyRestoreJobFinished(t *testing.T, ctx context.Context, f *framework.E2
 
 		return !restore.Status.FinishTime.IsZero()
 	}, polling.medusaRestoreDone.timeout, polling.medusaRestoreDone.interval, "restore didn't finish within timeout")
+
+	require.Eventually(func() bool {
+		secret := &corev1.Secret{}
+		err := f.Get(ctx, framework.NewClusterKey(restoreClusterKey.K8sContext, restoreClusterKey.Namespace, cassdcapi.CleanupForKubernetes(dcKey.Name)+"-superuser"), secret)
```

Comment:
Right, that's a good point. So I've made this change but now get errors on `CreateSingleMedusaJob` and `CreateSingleDseMedusaJob` because those clusters are named differently. I think this was the way I had it originally, but I changed it to reflect DC name to avoid errors on those tests.

I'll have to do something more complicated I think.